google/dev/jenkins_task_image_from_deb_repo.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+#
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script creates a Spinnaker GCP image from the Debian packages on Bintray.
+#
+# The following environment variables are expected to be populated:
+#
+# HOME [String] - Home directory for the user running this script.
+#
+# BUILDER_SERVICE_ACCOUNT [String] - Name of the service account used to build and store the image.
+# BUILDER_JSON_CREDENTIALS [String] - Path to the credentials file for $BUILDER_SERVICE_ACCOUNT.
+# BUILD_PROJECT [String] - Name of the GCP project the image is built in.
+# IMAGE_NAME [String] - Name of the resulting Spinnaker image.
+# IMAGE_PROJECT [String] - Name of the GCP project to publish the image to.
+# LOCAL_ZONE [String] - Zone used in this script to build the image. The final resulting image is independent of any zone.
+#
+# DEBIAN_REPO_URL [String] - Public bintray repository to install from. Defaults to the official spinnaker repository.
+# BINTRAY_KEY [String] - Key to use in the case where $DEBIAN_REPO_URL points to a private repository.
+# BINTRAY_USER [String] - Username to authenticate with in the case where $DEBIAN_REPO_URL points to a private repository.
+#
+# GITHUB_REPOSIROTY_BRANCH [String] - Branch within the repository this script should use.
+# GITHUB_REPOSITORY_OWNER [String] - Owner of the 'spinnaker' repository that contains the build_google_image.sh script used within this script.
+
+# Note that this script is GCP-specific -- it only produces GCP Spinnaker images currently.
+#
+# A minimal Jenkins job configuration to run this script:
+#
+# rm -rf spinnaker/
+# git clone https://github.com/spinnaker/spinnaker.git
+# ./spinnaker/dev/jenkins_task_image_from_deb_repo.sh
+
+
+function clone_spinnaker_with_owner() {
+  rm -rf spinnaker
```

Comment:
ok, but this is a very specialized use case. This is a generic script in the debian repository, but its intention is that it is running in the CWD of a jenkins server task workspace. Otherwise this rm -rf can ruin your day.

Maybe main should ensure that you are running this from a jenkins workspace?